src/strands/experimental/tools/tool_provider.py

```diff
@@ -0,0 +1,38 @@
+"""Tool provider interface."""
+
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Any, Sequence
+
+if TYPE_CHECKING:
+    from ...types.tools import AgentTool
+
+
+class ToolProvider(ABC):
```

Comment:
I'm aligned with ToolProviders - I'm a bit concerned about the use case of customers sharing ToolProviders between agents, I think that's a case that we want to support without creating spinning up duplicate servers.

After discussions offline with @dbschmigelski, I would propose the api(s) (naming TBD) to be roughly:

 - `register(agent: Agent)`
 - `unregister(agent: Agent)`
 - `load_tools()`

Where the tool-provider can then use registered things/agents as a way to know when to start/stop servers appropriately.